From 91056e585fde23ed2636fe473465aa3e6f429b91 Mon Sep 17 00:00:00 2001
From: "jie.chen" <jie.a.chen@intel.com>
Date: Thu, 17 Apr 2014 02:21:05 +0800
Subject: [PATCH] Do ApplyGpuDriverBugWorkarounds after
 InitializeOneOffForTests

gl_helper_unittests crashes on Android since ApplyGpuDriverBugWorkarounds calls
gl/egl API before it was bound by InitializeOneOffForTests
---
 content/common/gpu/client/gl_helper_unittest.cc | 2 --
 content/test/content_test_suite.cc              | 6 +++++-
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/content/common/gpu/client/gl_helper_unittest.cc b/content/common/gpu/client/gl_helper_unittest.cc
index 9e7af8c..acb1f59 100644
--- a/content/common/gpu/client/gl_helper_unittest.cc
+++ b/content/common/gpu/client/gl_helper_unittest.cc
@@ -27,7 +27,6 @@
 #include "content/common/gpu/client/gl_helper_scaling.h"
 #include "content/public/test/unittest_test_suite.h"
 #include "content/test/content_test_suite.h"
-#include "gpu/config/gpu_util.h"
 #include "media/base/video_frame.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "third_party/skia/include/core/SkBitmap.h"
@@ -1667,7 +1666,6 @@ int main(int argc, char** argv) {
 #if defined(OS_MACOSX)
   base::mac::ScopedNSAutoreleasePool pool;
 #endif
-  gpu::ApplyGpuDriverBugWorkarounds(CommandLine::ForCurrentProcess());
 
   content::UnitTestTestSuite runner(suite);
   base::MessageLoop message_loop;
diff --git a/content/test/content_test_suite.cc b/content/test/content_test_suite.cc
index ef1bfe9c..9aef770 100644
--- a/content/test/content_test_suite.cc
+++ b/content/test/content_test_suite.cc
@@ -10,6 +10,7 @@
 #include "content/public/common/content_client.h"
 #include "content/public/common/content_paths.h"
 #include "content/public/test/test_content_client_initializer.h"
+#include "gpu/config/gpu_util.h"
 #include "testing/gtest/include/gtest/gtest.h"
 #include "ui/base/resource/resource_bundle.h"
 
@@ -81,8 +82,11 @@ void ContentTestSuite::Initialize() {
   media::InitializeMediaLibraryForTesting();
   // When running in a child process for Mac sandbox tests, the sandbox exists
   // to initialize GL, so don't do it here.
-  if (!CommandLine::ForCurrentProcess()->HasSwitch(switches::kTestChildProcess))
+  if (!CommandLine::ForCurrentProcess()->HasSwitch(
+      switches::kTestChildProcess)) {
     gfx::GLSurface::InitializeOneOffForTests();
+    gpu::ApplyGpuDriverBugWorkarounds(CommandLine::ForCurrentProcess());
+  }
 #endif
   testing::TestEventListeners& listeners =
       testing::UnitTest::GetInstance()->listeners();
-- 
1.8.3.2

